File: Backend/Library/BJB.Dashboard.Library/Configuration/ConfigurationDictionary.cs
using System.Collections.Concurrent;

namespace BJB.Dashboard.Library.Configuration;

public class ConfigurationDictionary
{
    private static readonly ConcurrentDictionary<string, string> configMap = new ConcurrentDictionary<string, string>();

    public static void ReadConfig(string key, string value)
    {
        configMap.AddOrUpdate(key, value, (x, y) => y);
    }

    public static string? GetConfig(string code)
    {
        configMap.TryGetValue(code, out string? value);
        return value;
    }
}

File: Frontend/src/app/configuration/configuration-dictionary.ts
export class ConfigurationDictionary {
  private static readonly configMap = new Map<string, string>();

  static ReadConfig(key: string, value: string): void {
    this.configMap.set(key, value);
  }

  static GetConfig(key: string): string | undefined {
    return this.configMap.get(key);
  }
}
